README.md

```diff
@@ -430,6 +430,134 @@ public function getUser($credentials, UserProviderInterface $userProvider)
+    public function __construct(ClientRegistry $clientRegistry, EntityManagerInterface $entityManager, RouterInterface $router)
+    {
+        $this->clientRegistry = $clientRegistry;
+        $this->entityManager = $entityManager;
+        $this->router = $router;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function supports(Request $request): ?bool
+    {
+        // continue ONLY if the current ROUTE matches the check ROUTE
+        return $request->attributes->get('_route') === 'connect_facebook_check';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function authenticate(Request $request): PassportInterface
+    {
+        $credentials = $this->fetchAccessToken($this->clientRegistry->getClient('facebook_main'));
+
+        return new SelfValidatingPassport(
+            new UserBadge($credentials, function($credentials) {
+                /** @var FacebookUser $facebookUser */
+                $facebookUser = $this->clientRegistry->getClient('facebook_main')->fetchUserFromToken($credentials);
+
+                $email = $facebookUser->getEmail();
+
+                // 1) have they logged in with Facebook before? Easy!
+                $existingUser = $this->em->getRepository(User::class)->findOneBy['facebookId' => $facebookUser->getId()];
+
+                if ($existingUser) {
+                    return $existingUser;
+                }
+
+                // 2) do we have a matching user by email?
+                $user = $this->em->getRepository(User::class)->findOneBy(['email' => $email]);
+
+                // 3) Maybe you just want to "register" them by creating
+                // a User object
+                $user->setFacebookId($facebookUser->getId());
+                $this->em->persist($user);
+                $this->em->flush();
+
+                return $user;
+            })
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
+    {
+        // change "app_homepage" to some route in your app
+        $targetUrl = $this->router->generate('app_homepage');
+
+        return new RedirectResponse($targetUrl);
+    
+        // or, on success, let the request continue to be handled by the controller
+        //return null;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception): ?Response
+    {
+        $message = strtr($exception->getMessageKey(), $exception->getMessageData());
+
+        return new Response($message, Response::HTTP_FORBIDDEN);
+    }
+    
+    // ...
+}
+```
+### Step 2) Configuring the security
+Next, enable the new authenticator manager and then register your authenticator in `security.yaml` under the `guard` section:
+```yaml
+# app/config/packages/security.yaml
+security:
+    # ...
++   enable_authenticator_manager: true
++
+    firewalls:
+    	# ...
+        main:
```

Comment:
Looks like some bad indentation  here

Reply:
For this case it's strange, it looks good on PHPStorm, feel free to mark it as resolved if it's good on your side :)